File: backend/src/app/settlement/schemas.py
from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field

from app.constants import REMITTANCE_STATUS_SUCCEEDED


class GenerateRemittancesRequest(BaseModel):
    period_start: date
    period_end: date = Field(..., description="Inclusive end date of the settlement period.")
    attempt_status: str = Field(
        default=REMITTANCE_STATUS_SUCCEEDED,
        description="Desired remittance status for this settlement attempt.",
    )

File: backend/src/settlement/schemas.ts
import { IsDateString, IsNotEmpty } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';
export class GenerateRemittanceDto {
  @ApiProperty({
    description: 'Start date of the settlement period (ISO8601)',
    example: '2025-11-01',
    type: String,
  })
  @IsDateString()
  @IsNotEmpty()
  period_start: string;
 
  @ApiProperty({
    description: 'End date of the settlement period (ISO8601)',
    example: '2025-11-30',
    type: String,
  })
  @IsDateString()
  @IsNotEmpty()
  period_end: string;
}